src/porepy/models/fracture_damage.py

```diff
@@ -0,0 +1,326 @@
+                self.interface_displacement_variable,
+                self.contact_traction_variable,
+            ]
+        )
+
+
+class DamageHistoryEquation(pp.PorePyModel):
+    """Base class for damage history equations.
+
+    Sets up the damage history equation for the model. Since the equation considers the
+    full history, we reset the equation at each time step to include the new term.
+
+    """
+
+    damage_history_equation_name = "damage_history_equation"
+    """Name of the damage history equation."""
+
+    def set_equations(self):
+        super().set_equations()
+        fractures = self.mdg.subdomains(dim=self.nd - 1)
+        eq = self.damage_history_equation(fractures)
+        eq.set_name(self.damage_history_equation_name)
+        self.equation_system.set_equation(eq, fractures, {"cells": 1})
+
+    def before_nonlinear_loop(self):
+        """Reset the damage history equation to include new term.
+
+        This needs to be done *after* time manager is updated, which happens in the
+        super class method.
+
+        """
+        super().before_nonlinear_loop()
+        fractures = self.mdg.subdomains(dim=self.nd - 1)
+        self.equation_system._equations[self.damage_history_equation_name] = (
+            self.damage_history_equation(fractures)
+        )
+
+    def damage_history_equation(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Return the damage history equation.
+
+        Parameters:
+            subdomains: List of subdomains where the damage history is defined. Should
+                be of co-dimension one, i.e. fractures.
+
+        Returns:
+            Operator for the damage history equation.
+
+        """
+        raise NotImplementedError("Subclass must implement this method.")
+
+
+class AnisotropicHistoryEquation(DamageHistoryEquation):
+    r"""Anisotropic damage history equation.
+
+    The anisotropic damage history equation is given by
+
+    .. math::
+
+        h = \int{\tau} max(m_t \cdot \delta u_t, 0) \, d\tau,
+
+    where :math:`\delta u_t` is the tangential plastic displacement jump increment and
+    :math:`m_t` is the normalized tangential plastic displacement jump, while h is the
+    damage history variable.
+
+    """
+
+    damage_history: Callable[[list[pp.Grid]], pp.ad.Variable]
+    """Damage history variable on ad form."""
+
+    plastic_displacement_jump: Callable[[list[pp.Grid]], pp.ad.Operator]
+    """Method returning plastic displacement jump."""
+
+    def damage_history_equation(self, subdomains: list[pp.Grid]) -> pp.ad.Operator:
+        """Anisotropic damage history equation.
+
+        Parameters:
+            subdomains: List of subdomains where the damage history is defined. Should
+                be of co-dimension one, i.e. fractures.
+
+            Returns:
+                Operator for the anisotropic damage history equation.
+
+        """
+        # Get the tangential component of the plastic displacement jump.
```

Comment:
General comment regarding this function: The implementation looks correct to me, but since it involves much constitutive modeling with tacit assumptions in edge cases (cf. Heaviside function below), I would say that rather extensive documentation is needed. Before doing that, I suggest we talk through this once more on a blackboard, just to make sure we agree on all what are the choices to be made. This also regards the convolution integral and possible shortcuts to take there.

Reply:
Is it sufficiently documented now?